src/Stratis.Features.FederatedPeg/Wallet/MultiSigTransactions.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Features.FederatedPeg.Wallet
+{
+    public class MultiSigTransactions : ICollection<TransactionData>, ITransactionDataObserver
+    {
+        private readonly object lockObject;
```

Comment:
please add a comment about what data structures are protected by it. For each data structure add a remarks comment that it should be accessed under the lock

Reply:
It locks the whole class.